File: HomeWorks/src/Lesson6/Appraiser.java
package Lesson6;

public class Appraiser {

    private Robot theMostExpensiveRobot = new Robot();


    public Appraiser() {

    }

    public void whoIsTheMostExpensive (Robot[] robots) {

        Robot theMostExpensiveRobot = robots[0];
        for (Robot r : robots) {
            if (r.getPrice() >  theMostExpensiveRobot.getPrice()) {
                theMostExpensiveRobot = r;
            }
        }
        System.out.println("Самый дорогой робот из представленных - " + theMostExpensiveRobot.getName()
                + ". Его стоимость равна: " + theMostExpensiveRobot.getPrice());
    }
}

File: HomeWorkKotlin/src/Robot/Appraiser.kt
package Robot

class Appraiser {

    var theMostExpensiveRobot = Robot();

    fun whoIsTheMostExpensive (robots : Array<Robot>) {
        theMostExpensiveRobot = robots[0];
        for (item in robots) {
            if (item.price >  theMostExpensiveRobot.price) {
                theMostExpensiveRobot = item;
            }
        }
        println("Самый дорогой робот из представленных - ${theMostExpensiveRobot.robotName}. Его стоимость равна: ${theMostExpensiveRobot.price}")
    }
}


